### PR TITLE
ROSAENG-59908 | Add nightly terraform e2e test for rosa-boundary

### PR DESCRIPTION
**Pull request by theautoroboto**
## Summary
Adds nightly periodic test that deploys rosa-boundary AWS infrastructure via terraform, runs e2e tests, and tears down resources.

## Implementation Details
- **Step registry components** for provision, test, and teardown workflows
- Provisions deploy/network/ and deploy/regional/ terraform modules  
- Runs make test against provisioned infrastructure
- Uses cluster_profile: aws for credentials (no new secrets needed)
- **Scheduled**: 4 AM UTC daily (cron: 0 4 * * *)
- Teardown marked as best_effort to ensure cleanup attempts

## Testing
After merge, monitor first nightly execution at Prow dashboard to verify:
- Terraform provision succeeds
- Tests execute successfully  
- AWS resources properly cleaned up

## Files Added
- ci-operator/step-registry/rosa-boundary/nightly-provision/
- ci-operator/step-registry/rosa-boundary/nightly-e2e/
- ci-operator/step-registry/rosa-boundary/nightly-teardown/
- Generated Prow job: ci-operator/jobs/openshift-online/rosa-boundary/openshift-online-rosa-boundary-main-periodics.yaml

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

- Adds a nightly AWS Terraform end-to-end test for ROSA Boundary infrastructure.
- Provisions network and regional infrastructure, runs `make test`, and stores Terraform outputs in `${SHARED_DIR}`.
- Extracts and validates AWS credentials from `AWS_CONFIG_FILE`.
- Uses Terraform 1.15.8 and tears down resources in reverse order, while reporting teardown failures.
- Runs daily at 04:00 UTC through the generated Prow job with the AWS cluster profile.
- Sends nightly job failure and error notifications to `#wg-rosa-zero-operator-access`.

### CODE REVIEW

**Comment by theautoroboto** [comment]
/pj-rehearse periodic-ci-openshift-online-rosa-boundary-main-nightly-terraform-e2e

**Comment by theautoroboto** [comment]
/pj-rehearse periodic-ci-openshift-online-rosa-boundary-main-nightly-terraform-e2e

**Comment by theautoroboto** [comment]
/pj-rehearse periodic-ci-openshift-online-rosa-boundary-main-nightly-terraform-e2e

**Comment by theautoroboto** [comment]
/pj-rehearse periodic-ci-openshift-online-rosa-boundary-main-nightly-terraform-e2e

**Comment by theautoroboto** [comment]
/pj-rehearse periodic-ci-openshift-online-rosa-boundary-main-nightly-terraform-e2e

**Comment by theautoroboto** [comment]
/pj-rehearse periodic-ci-openshift-online-rosa-boundary-main-nightly-terraform-e2e

**Comment by theautoroboto** [comment]
/pj-rehearse periodic-ci-openshift-online-rosa-boundary-main-nightly-terraform-e2e

File: ci-operator/step-registry/rosa-boundary/nightly-provision/rosa-boundary-nightly-provision-commands.sh [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+# AWS credentials come from mounted secret via AWS_CONFIG_FILE env var
+# Extract credentials from AWS config file for Terraform
+if [[ -n "${AWS_CONFIG_FILE:-}" ]] && [[ -r "${AWS_CONFIG_FILE}" ]]; then
+  echo "Extracting AWS credentials from ${AWS_CONFIG_FILE}"
+
+  # Read credentials from the config file
+  # Format: [default]\naws_access_key_id = ...\naws_secret_access_key = ...
+  export AWS_ACCESS_KEY_ID=$(grep -A10 "\[default\]" "${AWS_CONFIG_FILE}" | grep "aws_access_key_id" | cut -d'=' -f2 | tr -d ' ')
+  export AWS_SECRET_ACCESS_KEY=$(grep -A10 "\[default\]" "${AWS_CONFIG_FILE}" | grep "aws_secret_access_key" | cut -d'=' -f2 | tr -d ' ')
```
[the remark this answers is not in the document]
Comment (theautoroboto) [reply ci-operator/step-registry/rosa-boundary/nightly-provision/rosa-boundary-nightly-provision-commands.sh]: no longer valid

**Comment by theautoroboto** [comment]
/pj-rehearse periodic-ci-openshift-online-rosa-boundary-main-nightly-terraform-e2e

**Comment by theautoroboto** [comment]
/pj-rehearse periodic-ci-openshift-online-rosa-boundary-main-nightly-terraform-e2e
